### PR TITLE
[MERGE] generic: set some fields as optional to improve and ease tree views usage

### PR DESCRIPTION
**Pull request by ppr-odoo**
PURPOSE

Purpose of this merge is to set the new attribute OPTIONAL = SHOW/Hide on
tree views so the user can easily hide/show the optional fields on the tree view.

SPECIFICATIONS

contacts (not used as main view but generic to all modules): 
  * phone: show
  * email: show
  * city: hide
  * state: hide
  * country: hide
  * next partner review: hide
  * level: hide
  * activation: hide
  * tags: hide
  * current membership status: hide
 
leads
  * created on: show
  * company name: hide
  * contact name: show
  * email: show
  * phone: show
  * city: show
  * state: hide
  * country: show
  * score: hide
  * assigned partner: show
  * sales team: show
  * salesperson: show
  * company: show
  * tags: hide
  * priority: hide

product template
  * internal reference: show
  * barcode: hide (to move after internal ref)
  * website: show
  * company: hide
  * sales price: show
  * cost: show:
  * product category: hide
  * product type: hide
  * quantity on hand: show
  * forecast quantity: show (rename to Forecasted Quantity)
  * unit of measure: show
  * ebay status: show

product product
  * internal reference: show
  * barcode: hide (to move after internal ref)
  * website: show
  * company: hide
  * public price: show (to rename in Sales price, to be consistent with product template form)
  * cost: show:
  * product category: hide
  * product type: hide
  * quantity on hand: show
  * forecast quantity: show
  * unit of measure: show

quotation
  * quotation date: show
  * website: show
  * sales team: hide (before sales person)
  * sales person: show
  * company: show
  * warehouse: hide
  * company: show
  * total: show
  * status: show
  * invoice status: hide
  * tags: hide

sales order
  * confirmation date: show
  * website: show
  * sales team: hide
  * salesperson: show
  * company: show
  * warehouse: hide
  * total: show
  * invoice status: show
  * tags: hide

helpdesk tickets
  * customer: show
  * assigned to: show
  * ticket type: hide
  * priority: hide
  * tags: hide
  * stage: show
  * company: show

RQP/purchase orders
  * vendor reference: hide
  * order date: show
  * company: show
  * scheduled date: show
  * Purchase rep: show
  * source document: show
  * untaxed: hide
  * total: show
  * status: show

Purchase agreement
  * purchase rep: show
  * company: show
  * ordering date: show
  * delivery date: hide
  * deadline: show
  * source document: show
  * status: show

Incoming products
  * date: hide (to rename into Creation date)
  * source document: show
  * status: show

Invoices & bills: 
  * invoice date: show
  * origin: show
  * payment reference: hide
  * company: show
  * sales team: hide
  * salesperson: show
  * due date: show
  * tax excluded: show
  * tax: hide
  * total: show
  * amount due: show
  * payment status: hide

Journal entries
  * partner: show
  * reference: show
  * company: show (after journal)

Journal items
  * reference: show
  * partner: show
  * analytic account: show
  * matching number: show
  * due date: hide

Analytic items
  * ref: hide
  * tags: hide (after analytic account)
  * financial account: hide
  * journal item: hide
  * partner: hide (before company)
  * company: show
  * product: hide
  * quantity: hide
  * unit of measure: hide
  * amount: show

Transfers (stock.picking)
  * from: show
  * to: show
  * partner: show
  * responsible: hide
  * operation type: hide
  * company: show
  * scheduled date: show
  * effective date: hide
  * source document: show
  * back order of: hide
  * status: show
  * batch picking: show
  * carrier: hide
  * tracking ref: hide
  * weight: hide
  * weight for shipping: hide
  * priority: hide

Inventory adjustment
  * location: hide (to add after inventory date)
  * company: show
  * accounting date: hide

Inventory report (stock.quant)
  * reserved: hide
  * company: show

Lot/serial
  * best before: hide (after creation date)
  * removal date: hide
  * end of life date: hide
  * alert date: hide
  * created on: show
  * internal ref: show

MO
  * deadline start: show
  * planning start date: hide (rename to planned start date)
  * product: show
  * bill of material: hide
  * source: show
  * responsible: hide
  * company: show
  * quantity: show
  * uom: show
  * routing: show
  * state: show
  * material availability: show

BoM
  * Reference: show
  * variant: show
  * version: show
  * quantity: show
  * unit of measure: show
  * bom type: hide
  * company: show
  * routing: show

Repair order:
  * product to repair: show
  * product quantity: hide
  * uom: hide
  * customer: show
  * delivery address: show
  * warranty expiration: show
  * location: hide
  * company: show
  * status: show

Quality alerts
  * product: show
  * product variant: hide
  * lot: hide
  * work center: show
  * team: show
  * responsible: show
  * check: show
  * tags: hide
  * root cause: hide
  * priority: hide
  * description: hide
  * date assigned: show
  * vendor: hide

Quality control points (and reorder tree view according to the list)
  * reference
  * title: show
  * product: show
  * product variant: hide
  * operation type: show
  * operation: hide
  * work order operation: hide
  * control type: hide
  * device: hide
  * type: show
  * team: show
  * responsible: hide
  * company: show

Quality checks (add the fields in the following order in the list view)
  * product: show
  * lot: show
  * picking: hide (rename to Transfer)
  * checked date: show
  * checked by: show
  * control point: hide
  * team: show
  * status: show

Project tasks
  * project: show
  * assigned to: show
  * customer: hide
  * tags: hide
  * planned hour: show
  * remaining hour: show
  * deadline: hide
  * stage: show
  * progress: show
  * company: show

LINKS

See task 1997453

### CODE REVIEW

File: addons/sale/views/sale_views.xml [hunk 1]
```diff
@@ -1082,6 +1087,17 @@
             </field>
         </record>
 
+        <record id="account_invoice_view_tree_inherit" model="ir.ui.view">
+            <field name="name">account.move.tree.inherit</field>
```
[the remark this answers is not in the document]
Comment (ppr-odoo) [reply addons/sale/views/sale_views.xml]: changes have done

File: addons/product_expiry/views/production_lot_views.xml [hunk 2]
```diff
@@ -35,4 +35,17 @@
         </field>
      </record>
 
+    <record id="view_production_lot_view_tree" model="ir.ui.view">
+        <field name="name">stock.production.lot.tree.inherit.stock</field>
```
Comment (dbeguin) [inline addons/product_expiry/views/production_lot_views.xml]: Guideline : as module2 = product_expiry here in name : model.view.tree.inherit.module2
this should become : `stock.production.lot.tree.inherit.product_expiry`

Same for all the other inherit here.

Comment (ppr-odoo) [reply addons/product_expiry/views/production_lot_views.xml]: @dbeguin changes  done.

**Comment by dbeguin** [comment]
LGTM.
@tde-banana-odoo

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-merge

**Pull request closed without merging**